### ISSUE TITLE
Overwrite Material Parameter in Families

### ISSUE THREAD

**Issue by Schindi11** [erfajo/OrchidForDynamo#7]
Hello @erfajo, I am still very happy with your nodes and i got an issue now where i am pretty sure you can help me. I am trying to overwrite the Material of all my Families because the Material Library changed - The changes are huge and not only affect some Appeareances, so a editing of the existing one will not work.  The Families are mainly created with Dynamo (Family Instance by Geometry) where the Material gets assigned, so there is a way to assign the Material and i would be very glad if you could help me with that.
I also already tried to get somehow the Parameter in a Project file but I can only access the Material Parameter when I run Dynamo in the Family - therefore I can only change 1 Family, quite useless.

My first try to do this was with your FamilyDocument.SetParameterByName or FamilyDocument.SetFormulaByName because I always use this nodes whenever i want to fill a parameter but in this case I cant find any way to assign a god damn new Material. Is there already a possibility to also change this Material in a Family Document using your nodes that I have overseen or is there the possibility to include this feature?
Would be also Glad to get a hint on how to solve it with another Workflow. The Material which should be assigned are available in an Excel Sheet with every Family Type and the matching Material (most of the Families are only 1 Solid).

Thanks a lot in advance

**Comment by erfajo** [comment]
You can change material parameters in families... try to see this
![parameter-test](https://user-images.githubusercontent.com/2844374/46175976-16f43000-c2ae-11e8-813b-946b78462c73.png)
you can find the .dyn file in the Samples

**Comment by Schindi11** [comment]
Hello,

I tried your script today and I always get an error when filling up the Test4 Material Parameter.
<img width="1121" alt="mat problem 1" src="https://user-images.githubusercontent.com/43644233/46281923-9f4d2c00-c55f-11e8-9aa9-0fe34cd2d744.png">
The project and the Family have both Glass Material inside if that is needed, I also tried with a few different Materials, always the same result.

Then i saved the Family that I tested and checked the Parameters. All the 4 Parameters are there but as described I am trying to fill the Standard Material Parameter which can be found in the Properties (Green Marked in Picture below). Is there a way to change this Material Parameter also? A workaround would be to create a Parameter, assign this one to the Material but I am also not sure if this is possible.
<img width="539" alt="mat problem 2" src="https://user-images.githubusercontent.com/43644233/46282130-34e8bb80-c560-11e8-8762-886b157da199.png">
Thanks in advance!

**Comment by erfajo** [comment]
First, this graph does not contain any scripting, is it purely a graph.
Next, Do you get a material id from the node "Material.ByName?
![image](https://user-images.githubusercontent.com/2844374/46290741-f3223a00-c58c-11e8-8e02-efb01cdd005b.png)

**Comment by Schindi11** [comment]
Hello, yeah that would fix the Problem with writing the Material to a Parameter.
Do you also have a solution for the Green Marked Problem in the Screenshot to directly write the Material in the Standard Properties? Thats my main problem :)

Thanks :)

**Comment by erfajo** [comment]
I dont understand what you are asking for!? do you work inside a family or a project? if it is a project, then the solution is the example used in this post. If it is in a project, then use the OOTB nodes for the comparative.

**Comment by Schindi11** [comment]
As described I am trying to change the Material of all ma Families we got so far.
My idea is to have them in a Folder, open a Project, Background open all the Families and overwrite the Material which is highlighted green in my screenshot above. I do not want to add a new Parameter like your Test4, I am trying to access the Standard Material Parameter in every Solid but I cant find a way to do that.

**Comment by erfajo** [comment]
Then you have got the solution... follow the example I gave in my first reply.

**Comment by Schindi11** [comment]
Hell erfajo, I updated now to Version 2.0.1 and tried to do that with Revit 2018. I still have the same error when i want to set the Parameter "Test4". I used exactly your Script, the Material is in the Project and in the Template, I get an Valid Material with ID but the node does not write anything.
<img width="940" alt="mat problem 1" src="https://user-images.githubusercontent.com/43644233/46412630-96e72380-c70e-11e8-86b8-3995e4c39ff3.png">

I would really appreciate you help to find a solution for that. I attached you the Graph, a Project File and also one dummy Family where I want to change the Material to whatever. This dummy File has now Cherry as Material - can be changed to any Material.
<img width="879" alt="mat problem 3" src="https://user-images.githubusercontent.com/43644233/46412637-9babd780-c70e-11e8-99c7-e49c68aac99d.png">

I just need the workflow on how to do this but my Dynamo skills seem to be not enough for this.
Thanks in advance for helping me out :)

**Comment by erfajo** [comment]
I have not tested it against built-in parameters. I had an understanding that those acted like all the other parameters, but it seems not to be the case.
I have no solution for this at the moment.
If you need to continue I can only help you if you add a parameter for materials.

**Comment by Schindi11** [comment]
Ok thank you, yeah it drives me crazy that I cant access them, all other Parameters are accessible but somehow it is possible because when creating Families with Srpings.FamilyInstance.byGeometry I can assign a Material which is written then in the System Parameter.

**Comment by erfajo** [comment]
Just to inform... the solution for associate a geometry material parameter is as below
![FormSetGet](https://user-images.githubusercontent.com/2844374/54761926-5a8f7d00-4bf3-11e9-9146-671a6a527e86.png)

**Comment by Amir-n-Jahed** [comment]
dear sir
when download your dyn file 
some windows say unresolved
what is your dynamo version
i use 2.0.3
if it's possible to you give me a solution

sincerely your's
amir.n.jahed@gmail.com

**Comment by erfajo** [comment]
that is the current version for 2018 and 2019. for Revit 2020 is 2.1.x.x the current version, and for 2017, 2018, 2019 is 1.3.4.x also available.
What has happened since this tread was written, is that I have updated some nodes, giving some of them a new name... and unfortunately do dynamo 2.x.x.x not provide a migration as the 1.3.4.x versions do. I have tried several time to address this for the dynamo team, but they dont seem to take that issue seriously. So I have no otion to help you there, however, I know that I have written a post in the dynamo forum describing how to "upgrade" --> https://forum.dynamobim.com/t/migrate-custom-nodes-to-zero-touch-nodes/31585/8
